File: chef_and_buld_invention/one.cpp
#include <iostream>

using namespace std;

int with_reminder_r(int N, int k, int r) {
    int n = N - 1;
    int ans = n/k;
    if (n%k >= r) {
        ++ans;
    }
    return ans;
}

int tc(int N, int k, int p) {
    int n = N - 1;
    int m = p % k - 1;
    int l = n % k;

    int ans = 1;
    ans += (m + 1) * (n / k);
    ans += min(l, m);

    ans += with_reminder_r(p + 1, k, p % k);

    return ans;
}

int main() {
    int t, N, p, k, ans;
    cin >> t;
    for (int i; i < t; i++) {
        cin >> N >> p >> k;
        ans = tc(N, k, p);
        cout << ans << endl;
    }
}

File: chef_and_buld_invention/one.py



def with_reminder_r(N, k, r) :
    n = N-1
    ans = n//k
    if n%k >= r :
        ans += 1;
    return ans


def with_max_rem_m(N, k, p):
    n = N-1
    m = p%k -1# maximum reminder
    l = n%k # last number reminder

    ans = 1
    ans += (m+1)* (n//k)
    ans += min(l,m)

    ans += with_reminder_r(p+1, k, p%k)

    return ans

N, k, r = 11, 4, 1
print(with_reminder_r(N, k, r))

N , k, p = 13, 4, 7
print(with_max_rem_m(N, k, p))